src/charm.py

```diff
@@ -307,35 +302,22 @@ def _set_password_action(self, event: ActionEvent) -> None:
             event.fail(msg)
             return
 
-        username = event.params.get("username", "sync")
-        if username not in CHARM_USERS:
-            msg = f"The action can be run only for users used by the charm: {CHARM_USERS} not {username}."
+        username = event.params.get("username", "")
+        new_password = event.params.get("password", generate_password())
+
+        if new_password in self.kafka_config.internal_user_credentials.values():
```

Comment:
This won't allow having the same password for both users. Not sure if it is intended or not, I'm just pointing it out.

Reply:
yes, probably I would also change the business logic. Also: are we really sure about defaulting to "" in the username above?

Reply:
It's intended, for security by separation of duty. If one pass runs away, the other is still safe (though they're both admin, so the improvement is minimal, it's still good practice). 

And defaulting to `""` is 'ok' I think. We enforce it to be an arg with my `config.py` changes, and the called `update_internal_user` will error out if `""`. I agree it's not explicit though. I'll add a check here.

Reply:
but I would say it is better to then make it:

```
username = event.params["username"]
```

If by any chance this value won't exist, an exception will occur as we expect (this should really never happen and it should not be handled, as it is a case not to happen by design. Beside, it is also clearer when reading the code

Reply:
I think I have an aversion to RuntimeFailures, but there are useful times, here's a good case, will do.